lighthouse-core/gather/driver/target-manager.js

```diff
@@ -5,70 +5,125 @@
  */
 'use strict';
 
-/**
-  * @fileoverview This class tracks multiple targets (the page itself and its OOPIFs) and allows consumers to
-  * listen for protocol events before each target is resumed.
-  */
-
+const EventEmitter = require('events').EventEmitter;
 const log = require('lighthouse-logger');
 const ProtocolSession = require('../../fraggle-rock/gather/session.js');
 
-/** @typedef {{target: LH.Crdp.Target.TargetInfo, session: LH.Gatherer.FRProtocolSession}} TargetWithSession */
+/**
+ * @typedef {{
+ *   target: LH.Crdp.Target.TargetInfo,
+ *   cdpSession: LH.Puppeteer.CDPSession,
+ *   session: LH.Gatherer.FRProtocolSession,
+ *   protocolListener: (event: unknown) => void,
+ * }} TargetWithSession
+ */
+
+// Add protocol event types to EventEmitter.
+/** @typedef {{'protocolevent': [LH.Protocol.RawEventMessage]}} ProtocolEventRecord */
+/** @typedef {new () => LH.Protocol.StrictEventEmitter<ProtocolEventRecord>} CrdpEventMessageEmitter */
+const ProtocolEventEmitter = /** @type {CrdpEventMessageEmitter} */ (EventEmitter);
 
-class TargetManager {
+/**
+ * Tracks targets (the page itself, its iframes, their iframes, etc) as they
```

Comment:
\+ workers? I know etc covers that...but still

Reply:
technically we [don't attach to those (yet)](https://github.com/GoogleChrome/lighthouse/pull/14122/files#diff-1d958bf77bb0afbc572b7eb91124d9adeec7802edfbf4fe59837ebf2e48a7e26R96) :)